src/solver/algorithm/strong/puzzle.rs

```diff
@@ -9,95 +9,233 @@ use anyhow::{Context, Result};
 
 use crate::database::volatile;
 use crate::database::{KVStore, Tabular};
-use crate::game::{Bounded, DTransition, GeneralSum, Playable, STransition};
+use crate::game::{Bounded, ClassicPuzzle, DTransition, Extensive, Game};
 use crate::interface::IOMode;
-use crate::model::{PlayerCount, Remoteness, State, Utility};
-use crate::solver::record::mur::RecordBuffer;
-use crate::solver::{RecordType, MAX_TRANSITIONS};
+use crate::model::SimpleUtility;
+use crate::model::{Remoteness, State};
+use crate::solver::error::SolverError::SolverViolation;
+use crate::solver::record::surcc::{ChildCount, RecordBuffer};
+use bitvec::{order::Msb0, prelude::*, slice::BitSlice, store::BitStore};
+use std::collections::{HashMap, HashSet, VecDeque};
 
-pub fn dynamic_solver<const N: usize, G>(game: &G, mode: IOMode) -> Result<()>
+pub fn dynamic_solver<G>(game: &G, mode: IOMode) -> Result<()>
 where
-    G: DTransition<State> + Bounded<State> + Playable<N> + GeneralSum<N>,
+    G: DTransition<State>
+        + Bounded<State>
+        + ClassicPuzzle
+        + Extensive<1>
+        + Game,
 {
     let mut db = volatile_database(game)
         .context("Failed to initialize volatile database.")?;
 
-    bfs(&mut db, game)
+    reverse_bfs_solver(&mut db, game)
         .context("Failed solving algorithm execution.")?;
 
     Ok(())
 }
 
-
-fn bfs<G, D>(game: &G, db: &mut D)
+fn reverse_bfs_solver<G, D>(db: &mut D, game: &G) -> Result<()>
```

Comment:
Thanks so much for coming up with this, please explain how this works later (a docstring would be a good way to do this). 

This is definitely a non-priority, and we can come up with a way to standardize how we communicate this along with the "solver requirements" later.